src/components/EmbeddedWalletProfileModal/index.tsx

```diff
@@ -0,0 +1,103 @@
+import { Button, Icon, Modal, Select } from "@stellar/design-system";
+import { useEffect, useState } from "react";
+
+import { CopyWithIcon } from "@/components/CopyWithIcon";
+import { STELLAR_EXPERT_URL } from "@/constants/envVariables";
+import { shortenAccountKey } from "@/helpers/shortenAccountKey";
+import { EmbeddedWalletReceiverContact } from "@/types";
+
+import "./styles.scss";
+
+type Props = {
+  isOpen: boolean;
+  onClose: () => void;
+  contact: EmbeddedWalletReceiverContact;
+  contractAddress: string;
+  currencies: string[];
+};
+
+export const EmbeddedWalletProfileModal = ({
+  isOpen,
+  onClose,
+  contact,
+  contractAddress,
+  currencies,
+}: Props) => {
+  const defaultCurrency = currencies[0];
+  const [currency, setCurrency] = useState(defaultCurrency);
+
+  useEffect(() => {
+    setCurrency(defaultCurrency);
+  }, [defaultCurrency]);
+
+  const formattedAddress = contractAddress
+    ? shortenAccountKey(contractAddress, 5, 5)
+    : "Wallet address unavailable";
+
+  const handleViewOnExplorer = () => {
+    if (!contractAddress) {
+      return;
+    }
+
+    const url = `${STELLAR_EXPERT_URL}/account/${contractAddress}`;
+    window.open(url, "_blank", "noopener");
+  };
+
+  return (
+    <Modal visible={isOpen} onClose={onClose}>
+      <Modal.Body>
+        <div className="EmbeddedWalletProfileModal__content">
+          <div className="EmbeddedWalletProfileModal__title">My profile</div>
+          <div className="EmbeddedWalletProfileModal__contact">{contact.value}</div>
+
+          <div className="EmbeddedWalletProfileModal__field">
+            <div className="EmbeddedWalletProfileModal__label">Stellar wallet address</div>
+            <div className="EmbeddedWalletProfileModal__addressRow">
+              <span className="EmbeddedWalletProfileModal__copyText">{formattedAddress}</span>
+              {contractAddress ? (
+                <div className="EmbeddedWalletProfileModal__copyAction">
+                  <CopyWithIcon textToCopy={contractAddress} iconSizeRem="1.25" doneLabel="Copied">
+                    <span />
+                  </CopyWithIcon>
+                </div>
+              ) : null}
```

Comment:
What if we render this conditionally:
1. If there is a contract address, wrap it in `CopyWithIcon` component
2. If there is no address, return only the text

The current approach has too many elements that are not needed, which makes it harder to follow what is going on.